src/adapter/src/coord/sequencer.rs

```diff
@@ -194,10 +199,20 @@ impl Coordinator {
                 ctx.retire(result);
             }
             Plan::CreateMaterializedView(plan) => {
-                let result = self
-                    .sequence_create_materialized_view(ctx.session_mut(), plan, resolved_ids)
-                    .await;
-                ctx.retire(result);
+                if enable_unified_optimizer_api {
+                    let result = self
+                        .sequence_create_materialized_view(ctx.session_mut(), plan, resolved_ids)
+                        .await;
+                    ctx.retire(result);
+                } else {
+                    // Allow while the introduction of the new optimizer API in
+                    // #20569 is in progress.
+                    #[allow(deprecated)]
+                    let result = self
+                        .sequence_create_materialized_view1(ctx.session_mut(), plan, resolved_ids)
```

Comment:
Should this be using the old or new API? `ship_dataflow` is being marked as deprecated, but in this branch the new `1` function is called. This logic is swapped?